libraries/AP_HAL/utility/BetterStream.cpp

```diff
@@ -29,8 +29,9 @@ int16_t AP_HAL::BetterStream::read()
     return b;
 }
 
-ssize_t AP_HAL::BetterStream::read(uint8_t *buffer, uint16_t count) {
-    uint16_t offset = 0;
+ssize_t AP_HAL::BetterStream::read(uint8_t *buffer, size_t count)
```

Comment:
IIRC we moved *to* this `uint16_t` interface because you didn't want callers to have very big numbers of bytes to play with and perhaps get into an essentially infinite loop.

Reply:
yes, good point, I've changed to uint16_t